### PR TITLE
10 bodkia add class master cab

### PR DESCRIPTION
**Pull request by bodkia22**

### CODE REVIEW

File: src/app/components/add-class-form/add-class-form.component.html [hunk 1]
```diff
@@ -10,6 +10,13 @@
 *ngIf="formIsOpen">
     <h1 class="title">add class</h1>
     <yps-form-field *ngFor="let field of fields" [field]="field"
-    [formControlName]="field.name"></yps-form-field>
+    [formControlName]="field.name">
+    </yps-form-field>
```
Comment (DmMov) [inline src/app/components/add-class-form/add-class-form.component.html]: Don't quite understand for moving [formControlName] to another line, it shoul be on the same line or you can move the other atributes to new lines.

File: src/app/services/teachers/teacherinfo.service.ts [hunk 2]
```diff
@@ -21,4 +22,9 @@ export class TeacherinfoService {
     this.httpOptions.loadHeaders();
     return this.http.get<IUser>(this.url + "/GetTeacherById", this.httpOptions.options);
   }
+
+  getTeachersToSelect=():Observable<ITeacherToSelect[]>=>{
+    this.httpOptions.loadHeaders();
+    return this.http.get<ITeacherToSelect[]>(this.url + "/GetTeacherBySchoolShort", this.httpOptions.options);
+  }
```
Comment (DmMov) [inline src/app/services/teachers/teacherinfo.service.ts]: Shit, think what I'm talking about.

File: src/app/components/add-class-form/add-class-form.component.ts [hunk 3]
```diff
@@ -13,44 +20,66 @@ import { validationHelper } from 'src/utils/helpers/validation-helper';
...
+          }
+        );
     }
   }
+
+  ngOnInit(): void {
+    this.getTeachersToSelectData();
+    this.httpOptionsService.loadHeaders();
+
+    this.form=this.formBuilder.group({
```
Comment (DmMov) [inline src/app/components/add-class-form/add-class-form.component.ts]: Shit as well

File: src/app/components/add-class-form/add-class-form.component.ts [hunk 4]
```diff
@@ -13,44 +20,66 @@ import { validationHelper } from 'src/utils/helpers/validation-helper';
...
+      label: "character",
+      placeholder: "enter character of the class",
+      name: "character",
+      errorMsg: null
     }
   ];
 
-  constructor(private formBuilder: FormBuilder) { }
+  
+  constructor(private formBuilder: FormBuilder, private http: HttpClient, private teacherService: TeacherinfoService,private httpOptionsService: HttpOptionsService) { }
```
Comment (DmMov) [inline src/app/components/add-class-form/add-class-form.component.ts]: Same

**Review by DmMov — APPROVED** [review]

**Review by shagi9 — APPROVED** [review]

**Review by Karonuk — APPROVED** [review]

**Pull request merged**